pkg/archive/archive_linux.go

```diff
@@ -2,29 +2,23 @@ package archive // import "github.com/docker/docker/pkg/archive"
 
 import (
 	"archive/tar"
-	"fmt"
-	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strings"
-	"syscall"
 
-	"github.com/containerd/continuity/fs"
 	"github.com/docker/docker/pkg/system"
-	"github.com/moby/sys/mount"
 	"github.com/pkg/errors"
 	"golang.org/x/sys/unix"
 )
 
 func getWhiteoutConverter(format WhiteoutFormat, inUserNS bool) tarWhiteoutConverter {
-	if format == OverlayWhiteoutFormat {
-		return overlayWhiteoutConverter{inUserNS: inUserNS}
+	if format == OverlayWhiteoutFormat && !inUserNS {
+		return overlayWhiteoutConverter{}
 	}
 	return nil
```

Comment:
For overlay in userns we never need a whiteoutconverter? Looking through the code, and I see there's to convert AuFS whiteouts to overlay, so was wondering if that would still be needed in that case

Reply:
Actually, this function will never be called when naiveDiffer is enabled. Let me know if I'm missing something.

Reply:
If UserNS doesn't support the non-naiveDiffer; should we produce an error before we're even hitting this? (marking the storage driver as not supported?)

Reply:
updated to raise an error